sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -270,12 +271,25 @@ trait CheckAnalysis extends PredicateHelper {
 
         operator match {
           case o if o.children.nonEmpty && o.missingInput.nonEmpty =>
+            val resolver = plan.conf.resolver
+            val attrsWithSameName = o.missingInput.filter(x =>
+              o.inputSet.exists(y => resolver(x.name, y.name)))
             val missingAttributes = o.missingInput.mkString(",")
-            val input = o.inputSet.mkString(",")
+            val availableAttributes = o.inputSet.mkString(",")
+            val repeatedNameHint = if (attrsWithSameName.size > 0) {
+              val commonNames = attrsWithSameName.map(_.name).mkString(",")
+              s"""\n|Attribute(s) `$commonNames` seem to appear in two
+                  |different datasets, with the same name."""
+            } else {
+              ""
+            }
 
             failAnalysis(
-              s"resolved attribute(s) $missingAttributes missing from $input " +
-                s"in operator ${operator.simpleString}")
+              s"""Some resolved attribute(s) are not present among the available attributes
+                |for a query.
+                |$missingAttributes is not in $availableAttributes.$repeatedNameHint
```

Comment:
Move `$repeatedNameHint` in next line as:
```scala
|$missingAttributes is not in $availableAttributes.
|$repeatedNameHint
```
So you can remove the new line above?

Reply:
Oh, thanks. I didn't see it

Reply:
This may add an empty line if `repeatedNameHint` is an empty string?

Reply:
Dang, I knew there was a reason for the weird new line back when I wrote this. I'm putting the `\n` back, do you have any idea that doesn't look as weird as this does, though?